dev_tools/nyaml2nxdl/README.md

```diff
@@ -0,0 +1,64 @@
+# YAML to NXDL converter and NXDL to YAML converter
+
+**NOTE: Please use python3.8 or above to run this converter**
+
+**Tools purpose**: Offer a simple way to edit NeXus definitions in YAML format. These can be NeXus application definitions or base classes. 
+Both forward (YAML -> NXDL.XML) and backward (NXDL.XML -> YAML) conversions are implemented.
+
+**How the tool works**:
+- nyaml2nxdl.py
+1. Reads the user-specified NeXus definition, either in YAML or XML format.
+2. If input is in YAML, it creates the definition in NXDL's XML format.
+   If input is an NXDL definition in XML format, it creates a corresponding YAML file.
+3. Optionally, if --append argument is given,
+   the XML or YAML input file is interpreted as an extension of the existing definition,
+   so instead of creating a new file from scratch, the original file is rather extended.
+
+```console
+user@box:~$ python nyaml2nxdl.py
+
+Usage: python nyaml2nxdl.py [OPTIONS]
+
+Options:
+   --input-file TEXT     The path to the input data file to read.
+   --append TEXT         Parse xml NeXus file and append to specified base class,
+                         write the base class name with no extension.
+   --check-consistency   Check consistency by generating another version of the input file.
+                         E.g. for input file: NXexample.nxdl.xml the output file
+                         NXexample_consistency.nxdl.xml.
+   --verbose             Additional std output info is printed to help debugging.
+   --help                Show this message and exit.
+
+```
+
+## Documentation
+
+**Rule set**: From transcoding YAML files we need to follow several rules.
+* Named NeXus groups, which are instances of NeXus classes especially base or contributed classes. Creating (NXbeam) is a simple example of a request to define a group named according to NeXus default rules. mybeam1(NXbeam) or mybeam2(NXbeam) are examples how to create multiple named instances at the same hierarchy level.
+* Members of groups so-called fields or attributes. A simple example of a member is voltage. Here the datatype is implied automatically as the default NeXus NX_CHAR type.  By contrast, voltage(NX_FLOAT) can be used to instantiate a member of class which should be of NeXus type NX_FLOAT.
+* And attributes of either groups or fields. The mark '\@' have to precede the name of attributes.
+* Optionality: For all fields, groups and attributes in `application definitions` are `required` by default, except anything (`recommended` or `optional`) mentioned.
```

Comment:
This is not clear.  Perhaps:

* In `application definitions`, all fields, groups, and attributes are `required` unless marked `recommended` or `optional`.

Reply:
suggestion has been put in.